### ISSUE TITLE
emailed party invitations don't give the invited member an invitation to the party

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9450]
When a person who doesn't have a Habitica account receives an emailed invitation to join Habitica, they are not given an invitation to the inviter's party.

Steps to reproduce:

1. Go to your party page.
2. Use the email invitation feature to send an email to an email address that does not currently have a Habitica account.
3. Click on the "join their Party on Habitica" link in the email.
4. Create a Habitica account using the exact email address used for the invitation.
5. The new account should have an invitation to your party but does not.

Here's a sample link from the email:
https://mandrillapp.com/track/click/30295795/habitica.com?p=eyJzIjoiMmanyothercharactersZDRlZjhjY1wiXX0ifQ

Part of the testing for the fix for this should ensure that when the invited player joins the party, the person who invited them should get the `invitedFriend` achievement.

**Comment by Alys** [comment]
labels updated

**Comment by Alys** [comment]
Part of the testing for the fix for this should ensure that when the invited player joins the party, the person who invited them should get the `invitedFriend` achievement.

**Comment by TheHollidayInn** [comment]
I'm not sure if this is working, but is the intention to redirect to `https://habitica.com?p=eyJzIjoiMmanyothercharactersZDRlZjhjY1wiXX0ifQ`

**Comment by TheHollidayInn** [comment]
Also, is there a different param for group invites? Here is the code:`if (req.query.groupInvite || req.query.partyInvite) {`

**Comment by Alys** [comment]
This is still happening. MorganLeFlay (4485e177-2203-4a07-b4e7-6ab5cda3ef60) reported in the Report a Bug guild that an invitation sent using their email address failed and I've confirmed it. I entered an email address that had not previously been used with Habitica into my party's "Invite by Email" form and received the expected "Help Alys Fight Monsters (and Bad Habits)!" email at that address. All three links within it were https://mandrillapp.com/track/click/30295795/habitica.com?p=eyJzIjoiSnVoam81Z2hidlRDWnVXTEtVQllPS1lreDhNIiwidiI6MSwicCI6IntcInVcIjozMDI5NTc5NSxcInZcIjoxLFwidXJsXCI6XCJodHRwczpcXFwvXFxcL2hhYml0aWNhLmNvbVwiLFwiaWRcIjpcIjAwZjFlYjg1OTdmODQ2MDhiNjRkODA3ZGQ5Nzk5YzhlXCIsXCJ1cmxfaWRzXCI6W1wiNWMzYTcyNjNkMDgzODdmNzMxOTE2NTRmNzI1ZTVjMGJjZDRlZjhjY1wiXX0ifQ

When I clicked on one of those links, I was redirected to exactly https://habitica.com and after registering using the same email address, I did not have an invitation to the party. After registering, I tried using the same link in the same browser while still logged in to Habitica and I still did not get a party invitation.
